### PR TITLE
Fix enotice when updating a custom group with is_multiple = 1

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
When updating a group with is_multiple = 1 there should be no change unless ['is_multiple']
is set but existing code is causing an e-notice in that case

Before
----------------------------------------
E-notice, new test fails

After
----------------------------------------
no e-notice, test passes

Technical Details
----------------------------------------

Comments
----------------------------------------
@michaelmcandrew perhaps you could look at this - kinda relates to yours

### CODE REVIEW

**Comment by michaelmcandrew** [comment]
hey @eileenmcnaughton - yes - very simliar. the new logic makes sense to me. I did some manual testing to get my head around it. works as I would expect so good to merge.

aside: not sure why we (before and now) we are entangling the need for an index update with the table name ($tableNameNeedingIndexUpdate) and not just using a boolean for $needsIndexUpdate and getting the table name from somewhere else. I would have thought that tableName should be disentangled from all those conditionals, but I am guessing you wanted to leave that part of the code alone (and I am resisting the temptation to increase the scope of your PR).

File: tests/phpunit/api/v3/CustomGroupTest.php [hunk 1]
```diff
@@ -300,11 +300,19 @@ public function testCustomGroupCreateGroup() {
       'help_post' => 'This is Post Help For Test Group 8',
     );
 
-    $customGroup = $this->callAPISuccess('custom_group', 'create', $params);
+    $customGroup = $this->callAPISuccess('CustomGroup', 'create', $params);
```
Comment (colemanw) [inline tests/phpunit/api/v3/CustomGroupTest.php]: These ought to be equivalent. Any reason why this change was made?

Comment (eileenmcnaughton) [reply tests/phpunit/api/v3/CustomGroupTest.php]: My understanding is that our preferred format for the entity is the camel case so it was for that reason

**Comment by seamuslee001** [comment]
(*CiviCRM Review Template WORD-1.1*)



* ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) Pass
* ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) Pass
* ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) Pass
* ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) Pass
* ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) Pass
* ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) Pass tested by Michael McAndrew
* ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) Pass
* ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) Pass
